PYI-540: Update Configuration Service to load Credential Issuers from…

Danomanic opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Update Configuration Service to load Credential Issuers from Parameter Store using new path method rather than YAML.

This change will call parameter store with the specified credentialIssuerId and retried all the parameters from within that path. At the moment, this is tokenUrl and credentialUrl. It then maps the response to a CredentialIssuerConfig object to be used by the CredentialIssuerHandler.

In this ticket I have also resoled an issue with Spotbugs not running in the lambda gradle build files by moving the task to the root build.gradle

<!-- Describe the changes in detail - the "what"-->

### Why did it change

To allow a better way to store the Credential Issuer config in Parameter Store.

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYI-540](https://govukverify.atlassian.net/browse/PYI-540)

## Checklists

### Environment variables or secrets

ENVIROMENT = dev/qa etc.

<!-- Delete if changes DO NOT include new environment variables or secrets -->
- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

kerrr commented on lib/build.gradle:
```diff
@@ -1,34 +1,33 @@
 plugins {
-    // Apply the java-library plugin for API and implementation separation.
-    id 'java-library'
+	// Apply the java-library plugin for API and implementation separation.
```
Can we stick with spaces please.

Danomanic replied:
Done

Commit:
PYI-540: Update Configuration Service to load Credential Issuers from Parameter Store using new path method rather than YMAL.

This change will call parameter store with the specified credentialIssuerId and retried all the parameters from within that path. At the moment, this is tokenUrl and credentialUrl. It then maps the response to a CredentialIssuerConfig object to be used by the CredentialIssuerHandler.

Co-authored-by: Sunil Dussoye <sunil.dussoye@digital.cabinet-office.gov.uk>
Signed-off-by: Daniel Pomfret <daniel.pomfret@digital.cabinet-office.gov.uk>
